From 1d7aae946435b5345b2290e54049618d43eeda4c Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Jun 2021 20:06:20 +0300
Subject: [PATCH] [SW-46517] greco: add dev specific register support

This patch implements lkd_dev_spec_regs function for secureboot
flow.

Change-Id: I91820e17cd0f1813e636e722b025c7eccf4c4fbd
---

diff --git a/arch/greco_arc_gnu/asic.c b/arch/greco_arc_gnu/asic.c
index 40e76e1..eabfbf6 100644
--- a/arch/greco_arc_gnu/asic.c
+++ b/arch/greco_arc_gnu/asic.c
@@ -9,10 +9,13 @@
 #include <psoc_gpio0_regs.h>
 #include <cpu_if_regs.h>
 
+#include <string.h>
 #include <bitops.h>
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
+#include <boot_desc.h>
+#include <hl_boot_if.h>
 #include <pci.h>
 
 #include "common.h"
@@ -89,3 +92,33 @@
 
 	hl_pcie_enable_access();
 }
+
+void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
+{
+	regs->cpu_pq_base_addr_low = mmCPU_IF_PQ_BASE_ADDR_LOW;
+	regs->cpu_pq_base_addr_high = mmCPU_IF_PQ_BASE_ADDR_HIGH;
+	regs->cpu_pq_length = mmCPU_IF_PQ_LENGTH;
+	regs->cpu_pq_init_status = mmCPU_IF_QUEUE_INIT;
+	regs->cpu_eq_base_addr_low = mmCPU_IF_EQ_BASE_ADDR_LOW;
+	regs->cpu_eq_base_addr_high = mmCPU_IF_EQ_BASE_ADDR_HIGH;
+	regs->cpu_eq_length = mmCPU_IF_EQ_LENGTH;
+	regs->cpu_eq_ci = mmCPU_IF_EQ_RD_OFFS;
+	regs->cpu_cq_base_addr_low = mmCPU_IF_CQ_BASE_ADDR_LOW;
+	regs->cpu_cq_base_addr_high = mmCPU_IF_CQ_BASE_ADDR_HIGH;
+	regs->cpu_cq_length = mmCPU_IF_CQ_LENGTH;
+	regs->cpu_pf_pq_pi = mmCPU_IF_PF_PQ_PI;
+	regs->cpu_boot_dev_sts0 = mmCPU_BOOT_DEV_STS0;
+	regs->cpu_boot_dev_sts1 = mmCPU_BOOT_DEV_STS1;
+	regs->cpu_boot_err0 = mmCPU_BOOT_ERR0;
+	regs->cpu_boot_err1 = mmCPU_BOOT_ERR1;
+	regs->cpu_boot_status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
+	regs->fw_upd_sts = mmUPD_STS;
+	regs->fw_upd_cmd = mmUPD_CMD;
+	regs->fw_upd_pending_sts = mmUPD_PENDING_STS;
+	regs->fuse_ver_offset = mmFUSE_VER_OFFSET;
+	regs->preboot_ver_offset = mmPREBOOT_VER_OFFSET;
+	regs->uboot_ver_offset = mmUBOOT_VER_OFFSET;
+	regs->hw_state = mmHW_STATE;
+	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
+	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
+}
